user_notification/models.py

```diff
@@ -8,16 +11,30 @@ class JobAlert(models.Model):
                                         on_delete=models.CASCADE,
                                         blank=True,
                                         null=True)
-    alert_message = models.TextField()
-    alert_frequency = models.CharField(max_length=20)
+    last_check_date = models.DateField(default=datetime.today)
+    frequency_in_days = models.PositiveIntegerField(default=1,
+                                                    validators=[MinValueValidator(1)])
     job_alert_type = models.CharField(max_length=50)
-    job_alert_scope = models.CharField(max_length=50)
-    job_alert_city = models.CharField(max_length=50)
-    job_alert_company_name = models.CharField(max_length=20)
+    job_alert_scope = models.CharField(max_length=50, choices=JobScope.choices, default=JobScope.UNSPECIFIED,
+                                       blank=True, null=True)
+    job_alert_city = models.CharField(max_length=50, blank=True, null=True)
+    job_alert_company_name = models.CharField(
+        max_length=20, blank=True, null=True)
 
     def __str__(self):
         return str(self.user_account_id)
 
+    # Calculate the date of the next alert
+    @classmethod
+    def calc_check_date(cls, last_check_date, frequency_in_days):
+        if isinstance(last_check_date, datetime):
```

Comment:
still - if frequency_in_days is a negative number would will be the output? 
You don't want to pass the function - need to raise an exception

Reply:
fixed